bootstrap/pkg/client/ksonnet/ksonnet.go

```diff
@@ -487,13 +487,15 @@ func (ksApp *ksApp) Init(resources kftypes.ResourceEnum, options map[string]inte
 func (ksApp *ksApp) initKs(config *rest.Config) error {
 	newRoot := path.Join(ksApp.KsApp.Spec.AppDir, ksApp.KsName)
 	ksApp.KsEnvName = kstypes.KsEnvName
-	k8sSpec := kftypes.GetServerVersion(kftypes.GetClientset(config))
+	// We hard code the K8s spec because we won't have a cluster to talk to when calling init.
```

Comment:
If we add serverVersion to the ClientSpec then it could be set by the platform and the package managers would use this value rather than attempting to contacting the server by using credentials. 

Similarly if the platform sets client credentials in the ClientSpec then the package managers would use this rather than $HOME/.kube/config's context. 

This would align with how the platform updates other values in ClientSpec like components, componentParameters and the package managers use this updated ClientSpec. 

I had discussed with @gabrielwen at the summit about this approach for fetching credentials. 
I have the PR for just using the ClientSpec about ready. (https://github.com/kubeflow/kubeflow/issues/2691)

Reply:
Only ksonnet needs to know the K8s version and that's so it can generate an appropriate k8s.libsonnet file from the corresponding swagger spec for the K8s API.

I think we can just hard code it.

* Only ksonnet needs to know the K8s API version; kustomize doesn't
  * ksonnet will likely be removed before the K8s API version is an issue
  * I suspect ksonnet might even generate a new lk8s lib file when a new environment is added (which we do later).

> Similarly if the platform sets client credentials in the ClientSpec then the package managers would use this rather than $HOME/.kube/config's context.

Why would client credentials need to be set in ClientSpec? How is that different then using a .kubeconfig file?

Reply:
Different platforms will set different credentials. If the package managers use the credentials set by the platform then we avoid package managers using .kube/config when they should be using the platform credentials.